src/blueapi/config.py

```diff
@@ -13,6 +13,26 @@ class StompConfig:
     port: int = 61613
 
 
+@dataclass  # TODO: Defer immediately to pika.ConnectionProperties?
+class AMQPConfig:
+    """
+    Config for connecting to the AMQP broker
+    """
+
+    host: str = "localhost"
+    port: int = 5672
+    userid: str = "guest"  # TODO: Config reads from env, so can be injected as Secret?
+    password: str = "guest"  # Can only use guest/guest on loopback address
+    virtual_host: str = "/"
+
+
+@dataclass
+class MessagingConfig:
+    impl: str = "stomp"
```

Comment:
To make it a bit more spring-esque, I suggest making implementations independent of eachother. For example:
```suggestion
    using: List[str] = ["stomp"]
```

and then when instantiating:
```python
for impl in config.using:
    if impl == "stomp":
        ...
    elif impl == "amqp":
        ...
```